samples/04.0-file-transfer/integrationtest/src/test/java/org/eclipse/dataspaceconnector/extension/sample/test/FileTransferSampleTest.java

```diff
@@ -0,0 +1,299 @@
+        // create object mapper instance
+        ObjectMapper mapper = new ObjectMapper();
+
+        // convert JSON file to map
+        DataRequest sampleDataRequest = mapper.readValue(transferJsonFile, DataRequest.class);
+
+        DataAddress newDataDestination = sampleDataRequest.getDataDestination().toBuilder().property("path", destinationFile.getAbsolutePath()).build();
+        DataRequest newSampleDataRequest = sampleDataRequest.toBuilder().contractId(contractAgreementId).dataDestination(newDataDestination).build();
+
+        return newSampleDataRequest;
+    }
+
+    /**
+     * Assert that the properties.config files are properly configured and all ports uniquely used.
+     * Relates to sample guidance section: Create the connectors.
+     */
+    void assertConfigPropertiesUniquePorts() {
+        String pathConsumerConfigProperties = new File(TestUtils.findBuildRoot(), CONSUMER_CONFIG_PROPERTIES_FILE_PATH).getAbsolutePath();
+        String pathProviderConfigProperties = new File(TestUtils.findBuildRoot(), PROVIDER_CONFIG_PROPERTIES_FILE_PATH).getAbsolutePath();
+
+        Properties consumerProperties = null;
+        Properties providerProperties = null;
+
+        try {
+            consumerProperties = readPropertiesFile(pathConsumerConfigProperties);
+            providerProperties = readPropertiesFile(pathProviderConfigProperties);
+        } catch (IOException e) {
+            Assertions.fail("Could not read properties files.");
+        }
+
+        List<String> portPropertyNames = List.of(
+                "web.http.port",
+                "web.http.data.port",
+                "web.http.ids.port"
+        );
+
+        HashMap<Integer, String> configuredPorts = new HashMap<>();
+
+        for (var portPropertyName : portPropertyNames) {
+            assertPortDefinedOnce(consumerProperties, portPropertyName, configuredPorts, "consumer");
+
+            assertPortDefinedOnce(providerProperties, portPropertyName, configuredPorts, "provider");
+        }
+    }
+
+    /**
+     * Helper method to assert that a given port number is existing only once in the given HashMap configuredPorts.
+     * If port number was not already defined it will be added to the HashMap configuredPorts with the given participant and property name.
+     */
+    private void assertPortDefinedOnce(Properties properties, String portPropertyName, HashMap<Integer, String> configuredPorts, String participantName) {
+
+        String propertyValue = properties.getProperty(portPropertyName);
+
+        if (propertyValue == null) return;
+
+        int portNumber = Integer.parseInt(propertyValue);
+
+        if (configuredPorts.containsKey(portNumber)) {
+            Assertions.fail(String.format("Port number '%d' defined multiple times.", portNumber));
+        }
+
+        configuredPorts.put(portNumber, String.format("%s/%s", participantName, portPropertyName));
```

Comment:
as you're not using the value it would be more efficient to use a Set than a Map.

Also you could use the return value of `put` to avoid an extra "round-trip" of `containsKey`

Reply:
Thanks for the catch, even though it is obsolete now. (The use of the value in the fail assertion above was missing to print out the already used property.)

I have removed the test for unique ports as it runs too late in the life cycle of the test. The EDC runtime extensions are instantiated before the tests run. Therefor a reuse of ports will fail before the tests could fail.